File: Java/Ejercicio-2/Ejercicio-2-4/Ejercicio_2_4_4/src/ejercicio_2_4_4/Ejercicio_2_4_4.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Main.java to edit this template
 */
package ejercicio_2_4_4;

/**
 *
 * @author daw1al13
 */
public class Ejercicio_2_4_4 {

    /**
     * @param args the command line arguments
     */
    public static void main(String[] args) {
        Wallet cartera1 = new Wallet(250,false);
        Wallet cartera2 = new Wallet(25,true);
        Wallet cartera3 = new Wallet(10,false);
        
        System.out.println(cartera1.available());
        System.out.println(cartera2.available());
        System.out.println(cartera3.available());
    
    }
    
}

File: Python/Ejercicio-2/Ejercicio-2-4/Ejercicio-2-4-4.py
# -*- coding: utf-8 -*-
# Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
# Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Main.java to edit this template

"""
@author Oscar Vaz Falagan
"""

class Wallet:
    """
    Clase Wallet simula una cartera con una cantidad de dinero y una tarjeta.
    """

    def __init__(self, money: float, card: bool):
        # Atributos de la cartera
        self.money = money
        self.card = card

    def available(self) -> str:
        """
        Método que devuelve un texto según el dinero y si tiene tarjeta o no.
        (La lógica exacta no estaba en el código original, 
        así que se infiere del contexto del ejercicio)
        """
        if self.money >= 100:
            return "Tienes suficiente dinero en efectivo."
        elif self.card:
            return "No tienes mucho dinero, pero tienes tarjeta."
        else:
            return "No tienes suficiente dinero ni tarjeta."


def main():
    # Creamos distintos objetos con distintos atributos para aplicarles un método
    cartera1 = Wallet(250, False)
    cartera2 = Wallet(25, True)
    cartera3 = Wallet(10, False)

    # Le pedimos que imprima el resultado del método available()
    # en base al objeto que uses
    print(cartera1.available())
    print(cartera2.available())
    print(cartera3.available())


if __name__ == "__main__":
    main()
